compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -703,7 +767,8 @@ class Typer extends Namer
     (index(stats), typedStats(stats, ctx.owner))
 
   def typedBlock(tree: untpd.Block, pt: Type)(implicit ctx: Context): Tree = track("typedBlock") {
-    val (exprCtx, stats1) = typedBlockStats(tree.stats)
+    val localCtx = ctx.retractMode(Mode.Pattern)
```

Comment:
Where did this `Mode.Pattern` come from? 
Was it from the pattern of something like?
```
bigFloat match {
  case 123_344_537_244_453E433 => ???
}
```

Reply:
Yes, exacty. We might make it user-accessible syntax though.